src/team_lib/nccl/xccl_nccl_collective.c

```diff
@@ -88,14 +88,14 @@ xccl_nccl_alltoall_start(xccl_tl_coll_req_t *request)
 
     NCCLCHECK(ncclGroupStart());
     for (peer = 0; peer < group_size; peer++) {
-      NCCLCHECK(ncclSend((void*)(sbuf + peer*data_size),
-                          data_size, ncclChar, peer,
-                          req->team->nccl_comm,
-                          req->team->stream));
-      NCCLCHECK(ncclRecv((void*)(rbuf + peer*data_size),
-                          data_size, ncclChar, peer,
-                          req->team->nccl_comm,
-                          req->team->stream));
+        NCCLCHECK(ncclSend((void*)(sbuf + peer*data_size),
+                           data_size, ncclChar, peer,
+                           req->team->nccl_comm,
+                           req->team->stream));
+        NCCLCHECK(ncclRecv((void*)(rbuf + peer*data_size),
+                           data_size, ncclChar, peer,
+                           req->team->nccl_comm,
+                           req->team->stream));
```

Comment:
@Sergei-Lebedev i see you don't use pairwise pattern here, is it intentional? All team ranks will first send to a single rank 0 - wouldn't it produce a congestion ?

Reply:
Yes, it's intentional. This linear schedule is what NCCL documentation suggests, also alltoall is implemented same way in nccl-tests performance benchmark